Sources/SwiftDocC/Model/TaskGroup.swift

```diff
@@ -93,6 +93,13 @@ struct ExtractLinks: MarkupRewriter {
                 guard let paragraph = item.child(at: 0) as? Paragraph,
                     paragraph.childCount >= 1 else { return true }
                 
+                // Check for trailing invalid content.
+                let containsInvalidContent = paragraph.children.dropFirst().contains(where: { child in
+                    let isComment = child is InlineHTML
+                    let isSpace = child is Text && child.format().trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
```

Comment:
I think `format()` could be a potentially expensive call. Since we only need to consider, Text – maybe we could use the `Text.string` property?

```suggestion
                    var isSpace = false
                    if let text = child as? Text {
                        isSpace = text.string.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
                    }
```

Reply:
We can also avoid doing the `isSpace` check if we already know that it's a comment.

```swift
if child is InlineHTML {
    return true // This is an approximation for HTML comments but will allow any inline HTML.
}
if let text = child as? Text {
    return text.string.allSatisfy { $0.isWhitespace || $0.isNewline }
}
return false
```

there are shorter more compact ways to write the same thing but in this case I think I'd prefer to spell it out a bit more. That also makes it easier to check for `@Comment` directives or to make the check for HTML comments more sophisticated in the future.